### DISCUSSION TITLE
Maintainance status

### DISCUSSION THREAD

**Discussion by zeenix** [woboq/qmetaobject-rs#276] [General]
Hi @ogoffart,

Long time, no see. :) I found myself needing QML+Rust integration and was very happy to see this project not only exists but that you're behind it. However, I see that the last release was almost an year ago and you're pretty active on the Slint project. So before I use it in a work project, I wanted to ask if you're aiming to maintain it?

I completely understand if the answer is "no" but it would be good to know in either case.

Thanks for your awesome work!

**Comment by ogoffart** [comment]
Passively maintained.

I still look at PR and issue, but i don't actively develop this repository.

It may work as is, and compared to alternative QML crates, has the advantage that it doesn't require to write C++ boiler plate.
But there is some work to do to keep it uptodate. This crate was implemented before Rust had (non-derive) procedural macro, so the API could now be improved using attribute macro and such.  Also it was before `std::pin::Pin`. There is some unsoundness in the API. And it doesn't support the latest feature from Qt.

The cxx-qt crate is maybe (from my impression) the most maintained Qt bindings at the moment, but they require to write some C++ code, which may not be a problem for your use case.

Otherwise, I can of course recommend to write your UI with [Slint](https://slint.rs) instead 😜

**Reply by zeenix** [reply]
Thanks for the very prompt and details response!

> The cxx-qt crate is maybe (from my impression) the most maintained Qt bindings at the moment, but they require to write some C++ code, which may not be a problem for your use case.

You caught me: I'm trying to avoid writing C++ code as much as possible here. :)

> Otherwise, I can of course recommend to write your UI with Slint instead stuck_out_tongue_winking_eye

I really wish I could but that decision is not up to me at all.

**Reply by ratijas** [reply]
Note that, however, neither of Qt<->Rust bindings projects have yet figured out how to compose QObject* pointers together, in a way that would make Rust compiler happy and keep programmers sane 🙃 so proceed with caution

**Reply by zeenix** [reply]
Thanks for the tip @ratijas !
